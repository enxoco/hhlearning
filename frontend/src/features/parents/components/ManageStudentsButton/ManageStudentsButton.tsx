import { chakra, IconButton, Tooltip } from "@chakra-ui/react";
import { EditIcon, AddIcon } from "@chakra-ui/icons";
import { Dispatch, SetStateAction } from "react";
import { Student } from "#/generated/graphql";
interface ButtonProps {
    id: string, portalId: string, email: string, students: Student[];
    setParentName: Dispatch<SetStateAction<string>>;
    setParentId: Dispatch<SetStateAction<string>>;
    lastName: string;
    onOpen: () => void;
  }
const Button = chakra(IconButton, {
    baseStyle: {
        color: "green",
        background: "white",
        borderRadius: "50%",
        padding: 1
    }
})
export default function ({ students, setParentName, setParentId, onOpen, lastName, id }: ButtonProps) {
    return (
        <div data-testid="manageChildrenButton">
        <Tooltip label={students.length ? "Manage Children" : "Add Children"}>
            <Button 
                aria-label={students.length ? "Manage Children" : "Add Children"} 
                icon={students.length ? <EditIcon boxSize="1.5em" /> : <AddIcon boxSize="1.5em" />} 
                _hover={{ background: "green", color: "white" }} 
                onClick={() => {
                    setParentName(lastName)
                    setParentId(id)
                    onOpen()
                }} 
            />
        </Tooltip>
        </div>
    )
}