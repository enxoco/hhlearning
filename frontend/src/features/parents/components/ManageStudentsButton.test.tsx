import '@testing-library/jest-dom'
import ManageStudentsButton from "./ManageStudentsButton";
import { render } from "@testing-library/react"
import { renderHook } from '@testing-library/react-hooks'
import { useState } from "react";
import { useDisclosure } from "@chakra-ui/react";

const { result: parentNameResult } = renderHook(() => useState(""));
const { result: parentIdResult } = renderHook(() => useState(""));
const { result: disclosureResult } = renderHook(() => useDisclosure());
test("renders add children button if no students found", async () => {
    const component = render(<ManageStudentsButton 
        id={"1"} 
        portalId={"abc"} 
        email={"test@test.com"} 
        students={[]} 
        setParentName={parentNameResult.current[1]} 
        lastName="Testerson" 
        setParentId={parentIdResult.current[1]} 
        onOpen={disclosureResult.current.onOpen} 
    />)

    expect(component.getByLabelText("Add Children")).toBeInTheDocument()
    expect(component.queryByLabelText("Manage Children")).not.toBeInTheDocument()
    expect(component).toMatchSnapshot()
})

test("renders manage children button if 1 or more students found", async () => {
    const component = render(<ManageStudentsButton 
        id={"1"} 
        portalId={"abc"} 
        email={"test@test.com"} 
        students={[{ id: "1", firstName: "Tom", lastName: "Test" }]} 
        setParentName={parentNameResult.current[1]} 
        lastName="Testerson" 
        setParentId={parentIdResult.current[1]} 
        onOpen={disclosureResult.current.onOpen} 
    />)

    expect(component.queryByLabelText("Add Children")).not.toBeInTheDocument()
    expect(component.getByLabelText("Manage Children")).toBeInTheDocument()
    expect(component).toMatchSnapshot()
})