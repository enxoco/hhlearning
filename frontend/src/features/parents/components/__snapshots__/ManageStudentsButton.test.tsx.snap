// Vitest Snapshot v1

exports[`renders add children button if no students found 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="manageChildrenButton"
      >
        <button
          aria-label="Add Children"
          class="chakra-button css-fmhdwd"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-196lzck"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="manageChildrenButton"
    >
      <button
        aria-label="Add Children"
        class="chakra-button css-fmhdwd"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-196lzck"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders manage children button if 1 or more students found 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="manageChildrenButton"
      >
        <button
          aria-label="Manage Children"
          class="chakra-button css-fmhdwd"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-196lzck"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="2"
            >
              <path
                d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
              />
              <path
                d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="manageChildrenButton"
    >
      <button
        aria-label="Manage Children"
        class="chakra-button css-fmhdwd"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-196lzck"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-width="2"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
